packages/libs/pactjs-test-project/src/transfer-create.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  IUnsignedTransaction,
+  Pact,
+  signWithChainweaver,
+} from '@kadena/pactjs';
+
+const sender: string = 'croesus';
+const receiver: string =
+  'k:554754f48b16df24b552f6832dda090642ed9658559fef9f3ee1bb4637ea7c94';
+// const sender: string = receiver;
+const senderNoK: string =
+  '2993f795d133fa5d0fd877a641cabc8b28cd36147f666988cacbaa4379d1ff93';
+// const senderNoK: string = sender.split('k:')[1];
+const amount: number = 1000.0;
+const data: Record<string, unknown> = {
+  ks: {
+    keys: ['554754f48b16df24b552f6832dda090642ed9658559fef9f3ee1bb4637ea7c94'],
+    pred: 'keys-all',
+  },
+};
+
+const unsignedTransaction: IUnsignedTransaction = Pact.modules.coin[
+  'transfer-create'
+](sender, receiver, () => 'read-keyset ks', amount)
```

Comment:
`createExp("read-keyset", "ks")` can be used here

Reply:
Yes that's one option. I'd rather have a list of classes that are exported from @kadena/pactjs that can be used as `guard`. Something like `type PactGuard = ReadKeyset | Keyset | OtherGuard`. Same goes for the other types. I'd like to have the flexiblity where we can map the javascript primitives dynamically to Pact Types (like PactGuard) so the consumer isn't required to use those classes.